plugin/trino-mysql/src/test/java/io/trino/plugin/mysql/AbstractTestMySqlTableStatistics.java

```diff
@@ -0,0 +1,461 @@
+        assertUpdate("DROP TABLE IF EXISTS " + tableName);
+        assertUpdate("" +
+                        "CREATE TABLE " + tableName + " AS " +
+                        "SELECT " +
+                        "    orderkey, " +
+                        "    if(orderkey % 3 = 0, NULL, custkey) custkey, " +
+                        "    if(orderkey % 5 = 0, NULL, orderpriority) orderpriority " +
+                        "FROM tpch.tiny.orders",
+                15000);
+        try {
+            gatherStats(tableName);
+            MaterializedResult statsResult = computeActual("SHOW STATS FOR " + tableName);
+            assertColumnStats(
+                    statsResult,
+                    new MapBuilder<String, Integer>()
+                            .put("orderkey", 15000)
+                            .put("custkey", 1000)
+                            .put("orderpriority", varcharNdvToExpected.apply(5))
+                            .build(),
+                    new MapBuilder<String, Double>()
+                            .put("orderkey", nullFractionToExpected.apply(0.0))
+                            .put("custkey", nullFractionToExpected.apply(1.0 / 3))
+                            .put("orderpriority", nullFractionToExpected.apply(1.0 / 5))
+                            .build());
+            assertThat(getTableCardinalityFromStats(statsResult)).isCloseTo(15000, withinPercentage(20));
+        }
+        finally {
+            assertUpdate("DROP TABLE " + tableName);
+        }
+    }
+
+    @Override
+    @Test
+    public void testAverageColumnLength()
+    {
+        throw new SkipException("MySQL connector does not report average column length");
+    }
+
+    @Override
+    @Test
+    public void testPartitionedTable()
+    {
+        throw new SkipException("Not implemented"); // TODO
+    }
+
+    @Override
+    @Test
+    public void testView()
+    {
+        String tableName = "test_stats_view";
+        executeInMysql("CREATE OR REPLACE VIEW " + tableName + " AS SELECT orderkey, custkey, orderpriority, comment FROM orders");
+        try {
+            assertQuery(
+                    "SHOW STATS FOR " + tableName,
+                    "VALUES " +
+                            "('orderkey', null, null, null, null, null, null)," +
+                            "('custkey', null, null, null, null, null, null)," +
+                            "('orderpriority', null, null, null, null, null, null)," +
+                            "('comment', null, null, null, null, null, null)," +
+                            "(null, null, null, null, null, null, null)");
+            // It's not possible to ANALYZE a VIEW in MySQL
+        }
+        finally {
+            executeInMysql("DROP VIEW " + tableName);
+        }
+    }
+
+    @Override
+    @Test
+    public void testMaterializedView()
+    {
+        throw new SkipException(""); // TODO is there a concept like materialized view in MySQL?
```

Comment:
Maybe we can remove TODO. There is no such thing in MySQL.

Reply:
This was supposed to document testing completeness , as required by superclass (`io.trino.plugin.jdbc.BaseJdbcTableStatisticsTest#testMaterializedView`)